### PR TITLE
Fixes #2942

### LINKED ISSUE #2942: Disenchanter disappears after max level cap
## :round_pushpin: Description (REQUIRED)

Hi, I want to report that when you put some Slimefun item with a higher enchantment that you set in your items.yml, for ex. I set 5:
```
  use-enchant-level-limit: true
  enchant-level-limit: 5

```
then a disenchanter & item disappear

## :bookmark_tabs: Steps to reproduce the Issue (REQUIRED)

1. Set enchant level limit to 5 and set true in use enchant level limit.
2. Put item with higher enchantment than 5 into the disenchanter
3. Now item & disenchanter disappear

## :bulb: Expected behavior (REQUIRED)

It probably should rather say that it's not possible to disenchant this item or something like this.

## :scroll: Server Log

```
[18:36:58] [Server thread/WARN]: [Slimefun] 
[18:36:58] [Server thread/WARN]: [Slimefun] An Error occurred! It has been saved as: 
[18:36:58] [Server thread/WARN]: [Slimefun] /plugins/Slimefun/error-reports/2021-04-04-18-36.err
[18:36:58] [Server thread/WARN]: [Slimefun] Please put this file on https://pastebin.com/ and report this to the developer(s).
[18:36:58] [Server thread/WARN]: [Slimefun] Bug Tracker: https://github.com/Slimefun/Slimefun4/issues
[18:36:58] [Server thread/WARN]: [Slimefun] 
[18:36:59] [Server thread/INFO]: _Fun0s_ issued server command: /back
[18:37:00] [Craft Scheduler Thread - 3779 - Slimefun/ERROR]: [Slimefun] X: -4,855 Y: 78 Z: -2,310 (AUTO_DISENCHANTER)
[18:37:00] [Craft Scheduler Thread - 3779 - Slimefun/ERROR]: [Slimefun] has thrown 4 error messages in the last 4 Ticks, the Block has been terminated.
[18:37:00] [Craft Scheduler Thread - 3779 - Slimefun/ERROR]: [Slimefun] Check your /plugins/Slimefun/error-reports/ folder for details.
[18:37:00] [Craft Scheduler Thread - 3779 - Slimefun/ERROR]: [Slimefun]  
[18:37:00] [Craft Scheduler Thread - 3779 - Slimefun/ERROR]: [Slimefun] An Exception was caught while ticking the Block Tickers Task for Slimefun vDEV - 869 (git 18139bb5)
java.util.ConcurrentModificationException: null
	at java.util.HashMap$HashIterator.nextNode(HashMap.java:1445) ~[?:1.8.0_281]
	at java.util.HashMap$KeyIterator.next(HashMap.java:1469) ~[?:1.8.0_281]
	at io.github.thebusybiscuit.slimefun4.implementation.tasks.TickerTask.tickChunk(TickerTask.java:144) ~[?:?]
	at io.github.thebusybiscuit.slimefun4.implementation.tasks.TickerTask.run(TickerTask.java:114) ~[?:?]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftTask.run(CraftTask.java:100) ~[patched_1.16.5.jar:git-Purpur-1048]
	at org.bukkit.craftbukkit.v1_16_R3.scheduler.CraftAsyncTask.run(CraftAsyncTask.java:58) ~[patched_1.16.5.jar:git-Purpur-1048]
	at com.destroystokyo.paper.ServerSchedulerReportingWrapper.run(ServerSchedulerReportingWrapper.java:22) ~[patched_1.16.5.jar:git-Purpur-1048]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_281]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_281]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_281]
```

## :open_file_folder: /error-reports/ Folder

https://pastebin.com/7LgBjJSz
https://pastebin.com/yywyLMuh
Two with auto & advanced disenchanters. BTW one is from InfinityExpansion IK.

## :compass: Environment (REQUIRED)

 - Server Software: Purpur
 - Minecraft Version:  1.16.5
 - Slimefun Version:  869

### PR DESCRIPTION
**Pull request by WalshyDev**
## Description
In the case of calling `Localization#getMessage(String)` there was no override in SlimefunLocalization meaning that no fallback was done.

Also fixed an issue with an invalid validation (https://github.com/Slimefun/Slimefun4/pull/2956/files#diff-b74eb692f379f300a12715d1a757f82e7aaeaa379276b8b528a5e78543a298beR155). Added some extra validates too.

## Changes
* Fixes #2942 
* Fixed incorrect Validate
* Added more Validates

## Related Issues
Resolves #2942 

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

### CODE REVIEW

**Review by TheBusyBiscuit — APPROVED** [review]
LGTM, so basically the fallback was in the overloaded method but not in the non-player one? :LUL:

Did you test this?

**Comment by WalshyDev** [comment]
> LGTM, so basically the fallback was in the overloaded method but not in the non-player one? :LUL:
> 
> Did you test this?

Yep exactly. This method wasn't used anywhere before the recent changes to max level. So, this only has 2 usages (enchanter + disenchanter). Makes sense why we didn't see it until now.

and yes, this was tested and works perfectly.

**Review by Sfiguz7 — DISMISSED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/services/localization/SlimefunLocalization.java [hunk 1]
```diff
@@ -48,55 +50,57 @@ protected SlimefunLocalization(@Nonnull SlimefunPlugin plugin) {
...
     public abstract Language getLanguage(@Nonnull Player p);
 
     /**
      * This method returns the default {@link Language} of this {@link Server}
-     * 
+     *
      * @return The default {@link Language}
      */
+    @Nonnull
     public abstract Language getDefaultLanguage();
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/services/localization/SlimefunLocalization.java:79]: Both of these are nullable.

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/services/localization/SlimefunLocalization.java [hunk 2]
```diff
@@ -229,6 +283,8 @@ public void sendMessage(CommandSender recipient, String key, boolean addPrefix)
     }
 
     public void sendActionbarMessage(@Nonnull Player player, @Nonnull String key, boolean addPrefix) {
+        Validate.notNull(player, "Player cannot be null!");
+
```
Comment (TheBusyBiscuit) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/services/localization/SlimefunLocalization.java:287]: Why no validation for the message key?

**Review by TheBusyBiscuit — COMMENTED** [review]
LGTM, just one question

**Comment by WalshyDev** [comment]
Happy RC day!

**Comment by TheBusyBiscuit** [comment]
I was literally just about to hit push on `stable`...

**Comment by WalshyDev** [comment]
Nuh! merge this first

**Comment by WalshyDev** [comment]
Sorry I was late on this, I was super caught up. Busy today too but just remembered it's RC day so cranked this out before I get ready

**Review by TheBusyBiscuit — APPROVED** [review]
Looks good to me.

@Sfiguz7 Approve pls

**Comment by TheBusyBiscuit** [comment]
> Sorry I was late on this, I was super caught up. Busy today too but just remembered it's RC day so cranked this out before I get ready

I really wanted this to be merged before RC too, I will delay the release until tomorrow so we can get a bit of time for potential bug reports inbetween this and the release.

**Review by Sfiguz7 — APPROVED** [review]

**Pull request merged**